setup.py

```diff
@@ -1,40 +1,100 @@
+##############################################################################
+#
+# Copyright (c) 2006 Zope Corporation and Contributors.
+# All Rights Reserved.
+#
+# This software is subject to the provisions of the Zope Public License,
+# Version 2.1 (ZPL).  A copy of the ZPL should accompany this distribution.
+# THIS SOFTWARE IS PROVIDED "AS IS" AND ANY AND ALL EXPRESS OR IMPLIED
+# WARRANTIES ARE DISCLAIMED, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF TITLE, MERCHANTABILITY, AGAINST INFRINGEMENT, AND FITNESS
+# FOR A PARTICULAR PURPOSE.
+#
+##############################################################################
+# This package is developed by the Zope Toolkit project, documented here:
+# http://docs.zope.org/zopetoolkit
+# When developing and releasing this package, please follow the documented
+# Zope Toolkit policies as described by this documentation.
+##############################################################################
+"""Setup for zope.locking package
+"""
+import os.path
+
 from setuptools import setup, find_packages
 
+
+def read(*rnames):
+    with open(os.path.join(os.path.dirname(__file__), *rnames)) as f:
+        return f.read()
+
+
+version = '3.0.0.dev0'
+
+
+tests_require = [
+    'transaction',
+    'zope.app.appsetup',
+    'zope.testing',
+    'zope.testrunner',
+    ]
+
+
 setup(
     name="zope.locking",
-    version="1.3dev",
+    version=version,
     license='ZPL 2.1',
     packages=find_packages('src'),
-    package_dir={'':'src'},
+    package_dir={'': 'src'},
     namespace_packages=['zope'],
     include_package_data=True,
-    install_requires = [
-        'setuptools',
-        'ZODB3',
+    install_requires=[
+        'BTrees',
+        'persistent',
         'pytz',
-        'zc.i18n',
-        'zope.app.generations',
-        'zope.app.keyreference',
-        'zope.app.publisher',
-        'zope.app.testing',
+        'setuptools',
         'zope.component',
         'zope.event',
-        'zope.formlib',
-        'zope.i18nmessageid',
+        'zope.generations',
         'zope.interface',
+        'zope.keyreference',
         'zope.location',
-        'zope.publisher',
         'zope.schema',
         'zope.security',
-        'zope.testing',
         ],
-    zip_safe = False,
-    description=open("README.txt").read(),
+    zip_safe=False,
+    tests_require=tests_require,
+    extras_require={'test': tests_require},
+    description=(
+        'Advisory exclusive locks, shared locks, and freezes '
+        '(locked to no-one).'),
     long_description=(
-        open('src/zope/locking/CHANGES.txt').read() +
-        '\n\n' +
-        open("src/zope/locking/README.txt").read()),
+        read('README.txt')
+        + '\n\n' +
+        read('src', 'zope', 'locking', 'README.txt')
+        + '\n\n' +
+        read('CHANGES.txt')
+        ),
```

Comment:
(I should've mentioned this earlier: would be nice to rename *.txt to *.rst for native GitHub rendering.)

Reply:
done